File: Python/reverse.py
def reverse(x):
    """
    :type x: int
    :rtype: int
    """
    ans = 0
    f =  x
    if x < 0:
        x= -x
    while x > 0:
       rem = x % 10 
       print rem
       ans = ans*10 + rem
       print ans
       x /= 10
    
    if ans > 2**31-1 or ans < -2**31+1:
        return 0
    if f < 0:
        ans = -ans
    else:
        return ans
        
print reverse(32768)

File: JavaScript/reverse.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    var sign = x > 0 ? 1 : -1;
    // num -> string -> array -> reverse
    var nx = x.toString().split('').reverse();
    // array -> string -> num
    var ans = parseInt(nx.join(''));
    // handle the overflow case however in JavaScript the Number.MAX_VALUE is way larger than this; 32 bit Max Int 2^31-1, 32 bit Min Int -2^31+1,
    if (ans > Math.pow(2, 31) - 1 || ans < -Math.pow(2, 31) + 1) {
        return 0;
    }
    return sign * ans;
};
